src/main/kotlin/net/atos/zac/enkelvoudiginformatieobject/EnkelvoudigInformatieObjectLockService.kt

```diff
@@ -0,0 +1,105 @@
+package net.atos.zac.enkelvoudiginformatieobject
+
+import jakarta.enterprise.context.ApplicationScoped
+import jakarta.inject.Inject
+import jakarta.persistence.EntityManager
+import jakarta.persistence.PersistenceContext
+import jakarta.transaction.Transactional
+import jakarta.transaction.Transactional.TxType.REQUIRED
+import jakarta.transaction.Transactional.TxType.SUPPORTS
+import kotlinx.coroutines.FlowPreview
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.onEach
+import kotlinx.coroutines.flow.retry
+import kotlinx.coroutines.flow.timeout
+import kotlinx.coroutines.runBlocking
+import net.atos.client.zgw.drc.DrcClientService
+import net.atos.client.zgw.zrc.ZRCClientService
+import net.atos.client.zgw.zrc.model.Zaak
+import net.atos.zac.enkelvoudiginformatieobject.model.EnkelvoudigInformatieObjectLock
+import net.atos.zac.util.UriUtil
+import nl.lifely.zac.util.AllOpen
+import nl.lifely.zac.util.NoArgConstructor
+import java.util.Optional
+import java.util.UUID
+import kotlin.time.Duration.Companion.milliseconds
+import kotlin.time.Duration.Companion.seconds
+
+@ApplicationScoped
+@Transactional(SUPPORTS)
+@AllOpen
+@NoArgConstructor
+class EnkelvoudigInformatieObjectLockService @Inject constructor(
+    private val drcClientService: DrcClientService,
+    private val zrcClientService: ZRCClientService
+) {
+    @PersistenceContext(unitName = "ZaakafhandelcomponentPU")
+    private lateinit var entityManager: EntityManager
+
+    @OptIn(FlowPreview::class)
```

Comment:
Looks ok. I also had a go at flow - retry but indeed it is a little clumsy it seems. Not sure if using this `timeout` preview thing is more readable however.. I find it a bit hard to see what is going on now.. I would personally prefer a more plain flow - retry I think.